opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/ProxyingSpan.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.opencensusshim;
+
+import com.google.errorprone.annotations.MustBeClosed;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Delegates <i>all</i> {@link Span} methods to some underlying Span via {@link
+ * ProxyingSpan#getProxied()}.
+ *
+ * <p>If not all calls are proxied, some, such as and in particular {@link
+ * Span#storeInContext(Context)} and {@link Span#makeCurrent()}, will use {@code this} instead of
+ * the proxied {@link Span} which betrays the expectation of instance fidelity imposed by the
+ * underlying otel mechanisms which minted the original {@link Span}, such as the otel javaagent.
+ *
+ * <p>This proxy class simplification allows the shim to perform its duties as minimally invasively
+ * as possible and itself never expose its own classes and objects to callers or recipients of calls
+ * from the shim.
+ *
+ * <p>This addresses the inconsistency where not all methods are appropriately delegated by exposing
+ * a single method, {@link ProxyingSpan#getProxied()}, to simplify and better ensure delegation and
+ * meeting expectations.
+ */
+// todo make this unnecessary
+@SuppressWarnings("UngroupedOverloads")
```

Comment:
is this just a matter of reordering methods?

Reply:
Yup. The extra grouping isn't required now that the test ensures all methods are delegated.